lib/value/src/kind/crud/remove.rs

```diff
@@ -49,15 +49,13 @@ impl Kind {
 
                     self.as_object_mut()
                         .map_or(CompactOptions::Never, |object| {
-                            match object.known_mut().get_mut(&Field::from(field.clone())) {
-                                None => {
-                                    // The modified value is discarded here (It's not needed)
-                                    &mut at_path_kind
-                                }
-                                Some(child) => child,
-                            }
-                            .remove_inner(&segments[1..], compact)
-                            .compact(object, field.clone(), compact)
+                            // The modified value is discarded here (It's not needed)
+                            object
+                                .known_mut()
+                                .get_mut(&Field::from(field.clone()))
+                                .map_or(&mut at_path_kind, |child| child)
```

Comment:
Is this not `.unwrap_or(&mut at_path_kind)`?

Reply:
Yep, good catch.